### ISSUE TITLE
Compute drift FutureWarning

### ISSUE THREAD

**Issue by caspervdw** [soft-matter/trackpy#474]
This should be looked at, from travis:
```
/home/travis/build/soft-matter/trackpy/trackpy/motion.py:289: FutureWarning: 'frame' is both a column name and an index level.
Defaulting to column but this will raise an ambiguity error in a future version
  dx = f_diff.loc[mask, pos_columns + ['frame']].groupby('frame').mean()
./home/travis/build/soft-matter/trackpy/trackpy/motion.py:289: FutureWarning: 'frame' is both a column name and an index level.
```

**Comment by sseyler** [comment]
I've now run into this error when calling `tp.compute_drift()` in the walkthrough. I'm currently running python 3.7.3, pandas 0.24.2, and trackpy 0.4.1. The full traceback is

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-62-a876d90e875a> in <module>
----> 1 d = tp.compute_drift(t1)

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/trackpy/motion.py in compute_drift(traj, smoothing, pos_columns)
    275         # swap the order for backwards compatibility
    276         pos_columns = guess_pos_columns(traj)
--> 277     f_sort = pandas_sort(traj, ['particle', 'frame'])
    278 
    279     # Compute the difference list of positions, particle, and frame columns.

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/trackpy/utils.py in _pandas_sort_post_017(df, by, *args, **kwargs)
    311 def _pandas_sort_post_017(df, by, *args, **kwargs):
    312     """Use sort_values() to sort a DataFrame"""
--> 313     return df.sort_values(*args, by=by, **kwargs)
    314 
    315 if is_pandas_since_017:

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/pandas/core/frame.py in sort_values(self, by, axis, ascending, inplace, kind, na_position)
   4709 
   4710             keys = [self._get_label_or_level_values(x, axis=axis)
-> 4711                     for x in by]
   4712             indexer = lexsort_indexer(keys, orders=ascending,
   4713                                       na_position=na_position)

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/pandas/core/frame.py in <listcomp>(.0)
   4709 
   4710             keys = [self._get_label_or_level_values(x, axis=axis)
-> 4711                     for x in by]
   4712             indexer = lexsort_indexer(keys, orders=ascending,
   4713                                       na_position=na_position)

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/pandas/core/generic.py in _get_label_or_level_values(self, key, axis)
   1699 
   1700         if self._is_label_reference(key, axis=axis):
-> 1701             self._check_label_or_level_ambiguity(key, axis=axis)
   1702             values = self.xs(key, axis=other_axes[0])._values
   1703         elif self._is_level_reference(key, axis=axis):

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/pandas/core/generic.py in _check_label_or_level_ambiguity(self, key, axis)
   1653                             label_article=label_article,
   1654                             label_type=label_type)
-> 1655             raise ValueError(msg)
   1656 
   1657     def _get_label_or_level_values(self, key, axis=0):

ValueError: 'frame' is both an index level and a column label, which is ambiguous.
```

My dumb workaround was to try simply renaming the index in the dataframe `t1`:

```python
t1_rename = t1.index.rename('frame_idx')
```

since it looked like the column named 'frame' (and not the index itself) was being used for sorting, but then I got the following error:

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-73-34d740b5a560> in <module>
----> 1 d = tp.compute_drift(t1_rename)

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/trackpy/motion.py in compute_drift(traj, smoothing, pos_columns)
    275         # swap the order for backwards compatibility
    276         pos_columns = guess_pos_columns(traj)
--> 277     f_sort = pandas_sort(traj, ['particle', 'frame'])
    278 
    279     # Compute the difference list of positions, particle, and frame columns.

~/Library/miniconda2/envs/trackpy/lib/python3.7/site-packages/trackpy/utils.py in _pandas_sort_post_017(df, by, *args, **kwargs)
    311 def _pandas_sort_post_017(df, by, *args, **kwargs):
    312     """Use sort_values() to sort a DataFrame"""
--> 313     return df.sort_values(*args, by=by, **kwargs)
    314 
    315 if is_pandas_since_017:

TypeError: sort_values() got an unexpected keyword argument 'by'
```

**Comment by nkeim** [comment]
This is now fixed, by #538 . We are overdue for another release, but in the meantime, I recommend [installing the development version](http://soft-matter.github.io/trackpy/v0.4.1/installation.html#latest-version-under-development)

**Comment by Resonanz** [comment]
Hey Nathan, I don't think this issue is fixed. I have tried today to run thru the Walkthrough on a Windows PC and a Mac and both fail with the same error. I installed the latest development version as suggested on both machines but to no avail.

`ValueError: 'frame' is both an index level and a column label, which is ambiguous.`
